### PR TITLE
Chore/doi

### PR DESCRIPTION
**Pull request by vutukuriajay2241**

### CODE REVIEW

File: projects/v2/src/app/components/doi/doi.component.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  error: Error = { hasError: false };
+  info: any;
+  constructor(
+    @Inject(MAT_BOTTOM_SHEET_DATA) public data: any,
+    public sheetRef: MatBottomSheetRef
+  ) {}
+
+  ngOnInit(): void {
+    this.loading = false;
+    this.data = this.data.map((item) => {
```
Comment (hrishikeshpaul) [inline projects/v2/src/app/components/doi/doi.component.ts:26]: What is happening here?

File: projects/v2/src/app/modules/root/root.component.ts [hunk 2]
```diff
@@ -80,13 +81,14 @@ export class RootComponent implements OnInit, OnDestroy{
...
   @Select(SheetState.getBottomSheetInfo) bottomSheetInfo$: Observable<SheetInfo>;
+  @Select(SheetState.getBottomSheetDOI) bottomSheetDOI$: Observable<DOI[]>;
 
   // Tree Observables
   @Select(TreeState.getTreeData) treeData$: Observable<any>;
   @Select(TreeState.getBottomSheetData) bsd$: Observable<any>;
   @Select(TreeState.getLinksData) links$: Observable<any>;
   @Select(TreeState.getBimodal) bm$: Observable<any>;
-  @Select(TreeState.getBiomarkerType) bmType$: Observable<string>;
+  // @Select(TreeState.getBiomarkerType) bmType$: Observable<string>;
```
Comment (hrishikeshpaul) [inline projects/v2/src/app/modules/root/root.component.ts:91]: Please base your branch to get this update

File: projects/v2/src/app/store/ui.state.ts [hunk 3]
```diff
@@ -86,7 +92,8 @@ export class UIStateModel {
     reportOpen: false,
     debugLogOpen: false,
     bottomSheetOpen: false,
-    compareOpen: false
+    compareOpen: false,
+    bottomSheetDOIOpen: false
```
Comment (hrishikeshpaul) [inline projects/v2/src/app/store/ui.state.ts:96]: What is the point of this, where are you using it? I only see SheetState.getDOI.... being used

File: projects/v2/src/app/store/ui.state.ts [hunk 4]
```diff
@@ -375,6 +383,44 @@ export class UIState {
...
+   * Action to close bottom sheet.
+   * Empty the bottom sheet data from the state
+   * Set the bottom sheet open variable to false
+   */
+   @Action(CloseBottomSheetDOI)
+   closeBottomSheetDOI({ getState, setState, dispatch }: StateContext<UIStateModel>) {
+     const state = getState();
+     dispatch(new UpdateBottomSheetData({}));
+     const view = this.store.selectSnapshot(TreeState.getVegaView);
+     if (Object.entries(view).length) { view.signal('path__click', {}); }
```
Comment (hrishikeshpaul) [inline projects/v2/src/app/store/ui.state.ts:416]: What is happening here?

**Review by hrishikeshpaul — COMMENTED** [review]
Well done! There are some reviews that I'd like to know the answers to!

File: projects/v2/src/app/components/doi/doi.component.ts (hunk 1 above)
Comment (vutukuriajay2241) [reply projects/v2/src/app/components/doi/doi.component.ts:26]: For each doi item the doi property starting with "doi: ". So I trimmed that part here.

File: projects/v2/src/app/store/ui.state.ts (hunk 3 above)
Comment (vutukuriajay2241) [reply projects/v2/src/app/store/ui.state.ts:96]: I need to remove that. I tried to do something😅

File: projects/v2/src/app/store/ui.state.ts (hunk 4 above)
Comment (vutukuriajay2241) [reply projects/v2/src/app/store/ui.state.ts:416]: I forgot to remove this, Will remove this as well

**Pull request merged**